R/data_coverage.R

```diff
@@ -0,0 +1,574 @@
+#' @title
+#' Check data coverage
+#'
+#' @description
+#' This function facilitates data coverage checks that analysts should perform
+#' during data exploration & cohort generation. Specifically, this function
+#' checks for "gaps" in the data timeline (e.g., are there any hospitals/time
+#' periods for which GEMINI did not receive any data at all?), and also provides
+#' more detailed insights into data volume & coverage by month and hospital
+#' (e.g., what percentage of encounters have an entry in a given table?).
+#' All checks performed by this function are applied at the table level (i.e.,
+#' missing values in individual columns are not considered and should be checked
+#' separately).
+#'
+#' @details
+#' `data_coverage` provides analysts with a tool to inform their decisions
+#' about which hospitals/time periods to include in their analyses,
+#' depending on the data tables of interest. For example, if a project relies on
+#' lab data (e.g., `mlaps` variable), users should carefully inspect lab data
+#' coverage (see `plot_coverage` below). If lab data coverage is high (e.g.,
+#' \>95% of `genc_ids` have an entry in the lab table), individual `genc_ids`
+#' may still not exist in the lab table (e.g., because lab testing was not
+#' indicated) and individual lab columns might still have missing values.
+#' Users should carefully consider how to handle these cases depending on the
+#' context of their analyses.
+#'
+#' @section Warning!:
+#'
+#' \itemize{
+#'    \item{Data coverage checks should generally be performed on the whole
+#' dataset, prior to applying any additional cohort inclusions/exclusions.}
+#'    \item{If you are an HPC4Health user and your datacut has been pre-filtered
+#'    based on certain inclusion/exclusion criteria (e.g., diagnosis codes),
+#'    please keep in mind that the coverage plots (see `plot_coverage`) may be
+#'    skewed in smaller/pre-filtered samples. Please reach out to the GEMINI
+#'    team if you need additional support.}
+#'    \item{Data coverage checks are particularly relevant for clinical data
+#'    tables (e.g., lab, pharmacy, radiology, transfusions, vitals, clinical
+#'    notes etc.).}
+#'    \item{This function should be used as a starting point for data coverage
+#'    checks, but users are advised to perform additional checks based on their
+#'    specific needs.}
+#' }
+#'
+#' @param dbcon (`DBIConnection`)\cr
+#' A database connection to any GEMINI database.
+#'
+#' @param cohort (`data.frame` or `data.table`)
+#' Cohort table with all relevant encounters of interest, where each row
+#' corresponds to a single encounter. Must contain the following columns:
+#' - `genc_id`: GEMINI Encounter ID
+#' - `hospital_num` | `hosital_id`: Hospital identifier
```

Comment:
*Super* minor, hosital_id instead of hospital_id

Reply:
Good catch! I just pushed a fix.